scripts/js/login.js

```diff
@@ -30,13 +30,20 @@ function wrongPassword(isError = false, isSuccess = false, data = null) {
     $("#error-message").text("");
     $("#error-hint").hide();
     $("#error-hint").text("");
-    if (data !== null && "error" in data.responseJSON && "message" in data.responseJSON.error) {
+    if (
+      // eslint-disable-next-line no-eqeqeq
+      data != null &&
+      "responseJSON" in data &&
+      "error" in data.responseJSON &&
+      "message" in data.responseJSON.error
+    ) {
       // This is an error, highlight both the password and the TOTP field
       isErrorResponse = true;
       // Check if the error is caused by an invalid TOTP token
       isInvalidTOTP = data.responseJSON.error.message === "Invalid 2FA token";
       $("#error-message").text(data.responseJSON.error.message);
-      if ("hint" in data.responseJSON.error && data.responseJSON.error.hint !== null) {
+      // eslint-disable-next-line no-eqeqeq
+      if ("hint" in data.responseJSON.error && data.responseJSON.error.hint != null) {
```

Comment:
This is unrelated to the bug but handles the case where we get back:
`{ responseJSON: { error: { ... } } }` (with no `hint` in `error`)

Reply:
> (with no `hint` in `error`)

When there is no `hint`, then the first part (`"hint" in data.responseJSON.error `) will be false.

Reply:
Thanks - I guess I was thinking `undefined` but that's not a use-case so I'll undo these and leave it at `responseJSON`